### ISSUE TITLE
Update to ES8

### ISSUE THREAD

**Issue by szaimen** [nextcloud/all-in-one#2332]
See https://www.elastic.co/guide/en/elastic-stack/8.7/upgrading-elastic-stack.html#prepare-to-upgrade

alternatively run `occ fulltextsearch:reset` to do a reindex and also reset the database

**Comment by szaimen** [comment]
Maxence said that the app still works fine even with ES7 so I will postpone the upgrade

**Comment by szaimen** [comment]
Apparently authentication is now required for ES? I got this error logged:
```
{
  "reqId": "2FuCif1dbJeirBlyZWNT",
  "level": 2,
  "time": "2023-06-06T14:59:37+00:00",
  "remoteAddr": "",
  "user": "--",
  "app": "fulltextsearch_elasticsearch",
  "method": "",
  "url": "--",
  "message": "401 Unauthorized: {\"error\":{\"root_cause\":[{\"type\":\"security_exception\",\"reason\":\"missing authentication credentials for REST request [/nextcloud-aio]\",\"header\":{\"WWW-Authenticate\":[\"Basic realm=\\\"security\\\" charset=\\\"UTF-8\\\"\",\"ApiKey\"]}}],\"type\":\"security_exception\",\"reason\":\"missing authentication credentials for REST request [/nextcloud-aio]\",\"header\":{\"WWW-Authenticate\":[\"Basic realm=\\\"security\\\" charset=\\\"UTF-8\\\"\",\"ApiKey\"]}},\"status\":401}",
  "userAgent": "--",
  "version": "26.0.2.1",
  "exception": {
    "Exception": "Elastic\\Elasticsearch\\Exception\\ClientResponseException",
    "Message": "401 Unauthorized: {\"error\":{\"root_cause\":[{\"type\":\"security_exception\",\"reason\":\"missing authentication credentials for REST request [/nextcloud-aio]\",\"header\":{\"WWW-Authenticate\":[\"Basic realm=\\\"security\\\" charset=\\\"UTF-8\\\"\",\"ApiKey\"]}}],\"type\":\"security_exception\",\"reason\":\"missing authentication credentials for REST request [/nextcloud-aio]\",\"header\":{\"WWW-Authenticate\":[\"Basic realm=\\\"security\\\" charset=\\\"UTF-8\\\"\",\"ApiKey\"]}},\"status\":401}",
    "Code": 401,
    "Trace": [
      {
        "file": "/var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/elasticsearch/elasticsearch/src/Client.php",
        "line": 172,
        "function": "setResponse",
        "class": "Elastic\\Elasticsearch\\Response\\Elasticsearch",
        "type": "->",
        "args": [
          [
            "GuzzleHttp\\Psr7\\Response"
          ],
          true
        ]
      },
      {
        "file": "/var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/elasticsearch/elasticsearch/src/Endpoints/Indices.php",
        "line": 385,
        "function": "sendRequest",
        "class": "Elastic\\Elasticsearch\\Client",
        "type": "->",
        "args": [
          [
            "GuzzleHttp\\Psr7\\Request"
          ]
        ]
      },
      {
        "file": "/var/www/html/custom_apps/fulltextsearch_elasticsearch/lib/Service/IndexService.php",
        "line": 144,
        "function": "delete",
        "class": "Elastic\\Elasticsearch\\Endpoints\\Indices",
        "type": "->",
        "args": [
          [
            "nextcloud-aio"
          ]
        ]
      },
      {
        "file": "/var/www/html/custom_apps/fulltextsearch_elasticsearch/lib/Platform/ElasticSearchPlatform.php",
        "line": 178,
        "function": "resetIndexAll",
        "class": "OCA\\FullTextSearch_Elasticsearch\\Service\\IndexService",
        "type": "->",
        "args": [
          [
            "Elastic\\Elasticsearch\\Client"
          ]
        ]
      },
      {
        "file": "/var/www/html/custom_apps/fulltextsearch/lib/Service/IndexService.php",
        "line": 601,
        "function": "resetIndex",
        "class": "OCA\\FullTextSearch_Elasticsearch\\Platform\\ElasticSearchPlatform",
        "type": "->",
        "args": [
          "all"
        ]
      },
      {
        "file": "/var/www/html/custom_apps/fulltextsearch/lib/Command/Reset.php",
        "line": 117,
        "function": "resetIndex",
        "class": "OCA\\FullTextSearch\\Service\\IndexService",
        "type": "->",
        "args": [
          "",
          ""
        ]
      },
      {
        "file": "/var/www/html/3rdparty/symfony/console/Command/Command.php",
        "line": 255,
        "function": "execute",
        "class": "OCA\\FullTextSearch\\Command\\Reset",
        "type": "->",
        "args": [
          [
            "Symfony\\Component\\Console\\Input\\ArgvInput"
          ],
          [
            "Symfony\\Component\\Console\\Output\\ConsoleOutput"
          ]
        ]
      },
      {
        "file": "/var/www/html/core/Command/Base.php",
        "line": 177,
        "function": "run",
        "class": "Symfony\\Component\\Console\\Command\\Command",
        "type": "->",
        "args": [
          [
            "Symfony\\Component\\Console\\Input\\ArgvInput"
          ],
          [
            "Symfony\\Component\\Console\\Output\\ConsoleOutput"
          ]
        ]
      },
      {
        "file": "/var/www/html/3rdparty/symfony/console/Application.php",
        "line": 1009,
        "function": "run",
        "class": "OC\\Core\\Command\\Base",
        "type": "->",
        "args": [
          [
            "Symfony\\Component\\Console\\Input\\ArgvInput"
          ],
          [
            "Symfony\\Component\\Console\\Output\\ConsoleOutput"
          ]
        ]
      },
      {
        "file": "/var/www/html/3rdparty/symfony/console/Application.php",
        "line": 273,
        "function": "doRunCommand",
        "class": "Symfony\\Component\\Console\\Application",
        "type": "->",
        "args": [
          [
            "OCA\\FullTextSearch\\Command\\Reset"
          ],
          [
            "Symfony\\Component\\Console\\Input\\ArgvInput"
          ],
          [
            "Symfony\\Component\\Console\\Output\\ConsoleOutput"
          ]
        ]
      },
      {
        "file": "/var/www/html/3rdparty/symfony/console/Application.php",
        "line": 149,
        "function": "doRun",
        "class": "Symfony\\Component\\Console\\Application",
        "type": "->",
        "args": [
          [
            "Symfony\\Component\\Console\\Input\\ArgvInput"
          ],
          [
            "Symfony\\Component\\Console\\Output\\ConsoleOutput"
          ]
        ]
      },
      {
        "file": "/var/www/html/lib/private/Console/Application.php",
        "line": 211,
        "function": "run",
        "class": "Symfony\\Component\\Console\\Application",
        "type": "->",
        "args": [
          [
            "Symfony\\Component\\Console\\Input\\ArgvInput"
          ],
          [
            "Symfony\\Component\\Console\\Output\\ConsoleOutput"
          ]
        ]
      },
      {
        "file": "/var/www/html/console.php",
        "line": 100,
        "function": "run",
        "class": "OC\\Console\\Application",
        "type": "->",
        "args": []
      },
      {
        "file": "/var/www/html/occ",
        "line": 11,
        "args": [
          "/var/www/html/console.php"
        ],
        "function": "require_once"
      }
    ],
    "File": "/var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/elasticsearch/elasticsearch/src/Response/Elasticsearch.php",
    "Line": 65,
    "message": "401 Unauthorized: {\"error\":{\"root_cause\":[{\"type\":\"security_exception\",\"reason\":\"missing authentication credentials for REST request [/nextcloud-aio]\",\"header\":{\"WWW-Authenticate\":[\"Basic realm=\\\"security\\\" charset=\\\"UTF-8\\\"\",\"ApiKey\"]}}],\"type\":\"security_exception\",\"reason\":\"missing authentication credentials for REST request [/nextcloud-aio]\",\"header\":{\"WWW-Authenticate\":[\"Basic realm=\\\"security\\\" charset=\\\"UTF-8\\\"\",\"ApiKey\"]}},\"status\":401}",
    "exception": [],
    "CustomMessage": "401 Unauthorized: {\"error\":{\"root_cause\":[{\"type\":\"security_exception\",\"reason\":\"missing authentication credentials for REST request [/nextcloud-aio]\",\"header\":{\"WWW-Authenticate\":[\"Basic realm=\\\"security\\\" charset=\\\"UTF-8\\\"\",\"ApiKey\"]}}],\"type\":\"security_exception\",\"reason\":\"missing authentication credentials for REST request [/nextcloud-aio]\",\"header\":{\"WWW-Authenticate\":[\"Basic realm=\\\"security\\\" charset=\\\"UTF-8\\\"\",\"ApiKey\"]}},\"status\":401}"
  },
  "id": "6480441754c0b"
}
```

**Comment by Zoey2936** [comment]
I use ES8 and it works without authentication...

**Comment by szaimen** [comment]
> I use ES8 and it works without authentication...

Oh really? Do you use it with Nextcloud?

Can you spot a problem what mistake I did when I tried it out and it did not work?

**Comment by Zoey2936** [comment]
> > I use ES8 and it works without authentication...
> 
> Do you use it with Nextcloud?

No, I use this compose file with ES v8: https://github.com/ZoeyVid/parsedmarc-dockerized/blob/develop/compose.yml

**Comment by szaimen** [comment]
> No, I use this compose file with ES v8: https://github.com/ZoeyVid/parsedmarc-dockerized/blob/develop/compose.yml

I see. https://github.com/ZoeyVid/parsedmarc-dockerized/blob/3da98f9dc3440e002a97b732de4d674e48cc742e/compose.yml#L21-L30 seems to be the solution then. Thanks a lot for sharing! :)

**Comment by szaimen** [comment]
This is now released with v6.4.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel
